Platform/NXP/Sabre_iMX6SX_1GB/Drivers/GraphicsOutputDxe/Lcdif.c

```diff
@@ -0,0 +1,220 @@
+    do
+    {
+        value32 = MmioRead32(IMX6SX_PHYSADDR_LCDIF2_RL);
+    } while (!(value32 & IMX6SX_LCDIF_RL_CLKGATE));
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_RL_CLR, IMX6SX_LCDIF_RL_SFTRST);
+    do
+    {
+        value32 = MmioRead32(IMX6SX_PHYSADDR_LCDIF2_RL);
+    } while (value32 & IMX6SX_LCDIF_RL_SFTRST);
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_RL_CLR, IMX6SX_LCDIF_RL_CLKGATE);
+    do
+    {
+        value32 = MmioRead32(IMX6SX_PHYSADDR_LCDIF2_RL);
+    } while (value32 & IMX6SX_LCDIF_RL_CLKGATE);
+}
+
+static
+VOID
+sConfigLCDIF(
+    IMX_DISPLAY_TIMING* Timing,
+    VOID* FrameBuffer
+    )
+{
+    UINT32 value32;
+    DEBUG((DEBUG_INFO,"sConfigLCDIF\n"));
+
+    // 24-bit hard coded
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_RL,
+        IMX6SX_LCDIF_RL_BYPASS_COUNT |
+        IMX6SX_LCDIF_RL_DOTCLK_MODE |
+        IMX6SX_LCDIF_RL_LCD_DATABUS_WIDTH_18 |
+        IMX6SX_LCDIF_RL_WORD_LENGTH_24 |
+        IMX6SX_LCDIF_RL_MASTER
+        );
+
+    // first three bytes out of every 32 contain valid data
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_CTRL1,
+        (0x07 << IMX6SX_LCDIF_CTRL1_BYTE_PACKING_FORMAT_SHL));
+
+    // Valid vertical and horizontal data
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_TRANSFER_COUNT,
+        (Timing->VActive << IMX6SX_LCDIF_TRANSFER_COUNT_V_SHL) |
+        Timing->HActive);
+
+    // V-Sync period
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_VDCTRL0,
+        IMX6SX_LCDIF_VDCTRL0_ENABLE_PRESENT |
+        IMX6SX_LCDIF_VDCTRL0_ENABLE_POL |
+        IMX6SX_LCDIF_VDCTRL0_VSYNC_PERIOD_UNIT |
+        IMX6SX_LCDIF_VDCTRL0_VSYNC_PULSE_UNIT |
+        Timing->VSync);
+
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_VDCTRL1, Timing->VActive + Timing->VBlank);
+
+    // H-Sync length and period
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_VDCTRL2,
+        (Timing->HSync << IMX6SX_LCDIF_VDCTRL2_HSYNC_PULSE_WIDTH_SHL) |
+        (Timing->HActive + Timing->HBlank));
+
+    // wait counts from start of sync to end of v/h line
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_VDCTRL3,
+        ((Timing->HBlank - Timing->HSyncOffset) << IMX6SX_LCDIF_VDCTRL3_HORZ_WAIT_CNT_SHL) |
+        ((Timing->VBlank - Timing->VSyncOffset)));
+
+    // valid data horizontal
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_VDCTRL4, Timing->HActive);
+
+    // frame buffer location
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_CUR_BUF, (UINT32)FrameBuffer);
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_NEXT_BUF, (UINT32)FrameBuffer);
+
+    // set clear
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_CTRL1_SET, IMX6SX_LCDIF_CTRL1_FIFO_CLEAR);
+
+    // turn on sync signals
+    value32 = MmioRead32(IMX6SX_PHYSADDR_LCDIF2_VDCTRL4);
+    value32 |= IMX6SX_LCDIF_VDCTRL4_SYNC_SIGNALS_ON;
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_VDCTRL4, value32);
+
+    // clear FIFO clear
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_CTRL1_CLR, IMX6SX_LCDIF_CTRL1_FIFO_CLEAR);
+
+    // set run bit
+    MmioWrite32(IMX6SX_PHYSADDR_LCDIF2_RL_SET, IMX6SX_LCDIF_RL_RUN);
+
+#if 0
```

Comment:
No `#if 0` code. Wrap it with an LCDIF_DEBUG define check.

Reply:
Fixed